pkg/river/internal/value/tag_cache.go

```diff
@@ -9,9 +9,9 @@ import (
 // tagsCache caches the river tags for a struct type. This is never cleared,
 // but since most structs will be statically created throughout the lifetime
 // of the process, this will consume a negligible amount of memory.
-var tagsCache = make(map[reflect.Type]objectFields)
+var tagsCache = make(map[reflect.Type]*objectFields)
 
-func getCachedTags(t reflect.Type) objectFields {
+func getCachedTags(t reflect.Type) *objectFields {
```

Comment:
Can this create duplicate tag trees? (likely reasonable but want for my understanding)

metrics.reciever "default" {
    targets = [ {
        url = "localhost"
     }]
}

exporter.redis "default" {
    targets = [ {
       url = "localhost"
    }]
}

type Target struct {
    URL string `river:"url"`
}

Running getCachedTags on metrics.receiver and exporter.redis will return two trees that have independent child nodes of Target? As opposed to two trees that point to a common Target `tree`?

Reply:
It doesn't dig down that deep. The tree it creates only covers the fields for a struct and how it maps to keys in an object representing that struct. 

if you have 

```go
type MetricsReceiverArguments struct {
   Targets []Target `river:"targets,attr"` 
}

type ExporterRedisArguments struct {
   Targets []Target `river:"targets,attr"` 
}
```

then `getCachedTags` returns two separate trees:

```go
// For MetricsReceiverArguments
&objectFields{
  fields: map[string]rivertags.Field{
    "targets": rivertags.Field{
       Name: []string{"targets"},
       Index: []int{0}, 
       Flags: rivertags.FlagAttr,
    },
  },
}

// For ExporterRedisArguments
&objectFields{
  fields: map[string]rivertags.Field{
    "targets": rivertags.Field{
       Name: []string{"targets"},
       Index: []int{0}, 
       Flags: rivertags.FlagAttr,
    },
  },
}
```

Notice that this isn't recursing into inner struct types. The "tree" only represents the hierarchy of keys within a given struct, so if you have a block name `a.b.c`, we hold leaves for `b` and `c` but that's it.